File: Unique_Morse_Code_Words/Answer.java
package umcw;

import java.util.ArrayList;
import java.util.HashSet;
import java.util.Set;

/**
 * Answer
 */
public class Answer {

  public static void main(String[] args) {

    String[] stary = {"gin", "zen", "gig", "msg"};

    Solution sol = new Solution();
    int rt = sol.uniqueMorseRepresentations(stary);

    System.out.println(rt);
  }
}

class Solution {

  private static String[] CODES = new String[] { ".-", "-...", "-.-.", "-..", ".", "..-.", "--.", "....", "..", ".---", "-.-", ".-..",
      "--", "-.", "---", ".--.", "--.-", ".-.", "...", "-", "..-", "...-", ".--", "-..-", "-.--", "--.." };
  public int uniqueMorseRepresentations(String[] words) {

    Set<String> set = new HashSet();
    for (String word : words) {

      StringBuilder str = new StringBuilder();
      for (char c: word.toCharArray()){
         str.append(CODES[c - 'a']);
      }
      set.add(str.toString());
    }

    return set.size();
  }
}

File: Unique_Morse_Code_Words/answer.py
class Solution:
    def uniqueMorseRepresentations(self, words) -> int:
        CODES = [".-", "-...", "-.-.", "-..", ".", "..-.",
                 "--.", "....", "..", ".---", "-.-", ".-..",
                 "--", "-.", "---", ".--.", "--.-", ".-.", "...",
                 "-", "..-", "...-", ".--", "-..-", "-.--", "--.."]
        collection = []
        for word in words:
            code = ""
            for char in word:
                code = code + CODES[ord(char) - ord("a")]
            collection.append(code)

        return len(set(collection))

sol = Solution()
stary = ["gin", "zen", "gig", "msg"]

print(sol.uniqueMorseRepresentations(stary))
